memory_usage.go

```diff
@@ -20,16 +20,38 @@ type MemoryUsage struct {
 	CacheTotal uint64
 }
 
+type DBForMemoryUsage interface {
+	getDBforMemoryUsage() *C.rocksdb_t
+}
```

Comment:
not: something like `NativeDB` and `GetNativeDB` would better align with the existing naming convention in `gorocksdb`, see the existing types and functions e.g. `nativeComparator`, `NewNativeComparator`.